setup.py

```diff
@@ -18,7 +18,7 @@
 setup(
     name="spinegeneric",
     version=spinegeneric.__version__,
-    python_requires=">=3.7,<3.12",
+    python_requires=">=3.9,<=3.13",
```

Comment:
I recently happened to see [a giant post against upper version bounds](https://iscinumpy.dev/post/bound-version-constraints/) in `python_requires`, so:
```suggestion
    python_requires=">=3.9",
```